samples/webassembly/Triangle/Triangle.cpp

```diff
@@ -0,0 +1,276 @@
+#endif
+
+#if defined(__cpp_lib_format)
+#include <format>
+#define IGL_FORMAT std::format
+#else
+#include <fmt/core.h>
+#define IGL_FORMAT fmt::format
+#endif // __cpp_lib_format
+
+std::string codeVS = R"(#version 300 es
+
+precision highp float;
+
+out vec3 vColor;
+const vec2 pos[3] = vec2[3](
+	vec2(-0.6, -0.4),
+	vec2( 0.6, -0.4),
+	vec2( 0.0,  0.6)
+);
+const vec3 col[3] = vec3[3](
+	vec3(1.0, 0.0, 0.0),
+	vec3(0.0, 1.0, 0.0),
+	vec3(0.0, 0.0, 1.0)
+);
+void main() {
+	gl_Position = vec4(pos[gl_VertexID], 0.0, 1.0);
+	vColor = col[gl_VertexID];
+}
+)";
+
+const char* codeFS = R"(#version 300 es
+
+precision highp float;
+
+in vec3 vColor;
+layout (location=0) out vec4 oColor;
+void main() {
+	oColor = vec4(vColor, 1.0);
+}
+)";
+
+using namespace igl;
+
+GLFWwindow* window_ = nullptr;
+int width_ = 0;
+int height_ = 0;
+
+std::unique_ptr<IDevice> device_;
+std::shared_ptr<ICommandQueue> commandQueue_;
+RenderPassDesc renderPass_;
+std::shared_ptr<IFramebuffer> framebuffer_;
+std::shared_ptr<IRenderPipelineState> renderPipelineState_Triangle_;
+
+static bool initWindow(GLFWwindow** outWindow) {
+  if (!glfwInit()) {
+    return false;
+  }
+
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);
+  glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_COMPAT_PROFILE);
+  glfwWindowHint(GLFW_VISIBLE, true);
+  glfwWindowHint(GLFW_DOUBLEBUFFER, true);
+  glfwWindowHint(GLFW_SRGB_CAPABLE, true);
+  glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_API);
+  glfwWindowHint(GLFW_RESIZABLE, GLFW_TRUE);
+
+  GLFWwindow* window = glfwCreateWindow(800, 600, "WebGL Triangle", nullptr, nullptr);
+
+  if (!window) {
+    glfwTerminate();
+    return false;
+  }
+
+  glfwSetErrorCallback([](int error, const char* description) {
+    printf("GLFW Error (%i): %s\n", error, description);
+  });
+
+  glfwSetKeyCallback(window, [](GLFWwindow* window, int key, int, int action, int) {
+    if (key == GLFW_KEY_ESCAPE && action == GLFW_PRESS) {
+      glfwSetWindowShouldClose(window, GLFW_TRUE);
+    }
+  });
+
+  // @lint-ignore CLANGTIDY
+  glfwSetWindowSizeCallback(window, [](GLFWwindow* /*window*/, int width, int height) {
+    printf("Window resized! width=%d, height=%d\n", width, height);
+    width_ = width;
+    height_ = height;
+  });
+
+  glfwGetWindowSize(window, &width_, &height_);
+
+  if (outWindow) {
+    *outWindow = window;
+  }
+
+  return true;
+}
+
+static void initIGL() {
+  // create a device
+  {
+    auto ctx =
+        std::make_unique<igl::opengl::webgl::Context>(igl::opengl::RenderingAPI::GLES3, "#canvas");
+    device_ = std::make_unique<igl::opengl::webgl::Device>(std::move(ctx));
+
+    IGL_ASSERT(device_);
+  }
+
+  // Command queue: backed by different types of GPU HW queues
+  CommandQueueDesc desc{CommandQueueType::Graphics};
+  commandQueue_ = device_->createCommandQueue(desc, nullptr);
+
+  // first color attachment
+  for (auto i = 0; i < kNumColorAttachments; ++i) {
```

Comment:
I recommend removing this for the sake of simplicity and keeping only 1 color attachment.